internal/observability/exporter/prometheus/prometheus.go

```diff
@@ -0,0 +1,92 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package prometheus provides a prometheus exporter.
+package prometheus
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"contrib.go.opencensus.io/exporter/prometheus"
+)
+
+var (
+	instance *exporter
+	once     sync.Once
+)
+
+type prometheusOptions = prometheus.Options
+
+type Prometheus interface {
+	Start(ctx context.Context) error
+	Stop(ctx context.Context)
+	Exporter() *prometheus.Exporter
+}
+
+type exporter struct {
+	exporter *prometheus.Exporter
+	options  prometheusOptions
+}
+
+func New(opts ...PrometheusOption) (Prometheus, error) {
+	po := new(prometheusOptions)
+
+	for _, opt := range append(prometheusDefaultOpts, opts...) {
+		err := opt(po)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	ex, err := prometheus.NewExporter(*po)
+	if err != nil {
+		return nil, err
+	}
+
+	e := exporter{
+		exporter: ex,
+		options:  *po,
+	}
+
+	once.Do(func() {
+		instance = &e
```

Comment:
same, static method configuration should be avoided unless absolutely necessary

Reply:
Do you have any ideas to pass NewHTTPHandler to internal/servers/starter.go ?
I don't know how to do it without static method...
https://github.com/vdaas/vald/pull/131/files#diff-0781766d6ccc6cdf65ab3c461b8b1e0aR168

Reply:
it may be an absolutely necessary case, so i refactored it like that:
https://github.com/vdaas/vald/pull/131/commits/e8d65f624098ba2399d12d0b32f3bbb3a3f858fd#diff-2ddd64cb7d3de612c03b1f4d709f8a0dR95